### ISSUE TITLE
TermAnnotation

### ISSUE THREAD

**Issue by pnrobinson** [monarch-initiative/phenol#345]
This interface (in org.monarchinitiative.phenol.ontology.data) currently assumes that both the Item (e.g., a gene or a disease) and the annotation (e.g., a Gene Ontology or HPO term) can be represented by a TermId.
It would be better to have the interface be generic because we might want to use other domain objects for annotation, e.g., an AccessionNumber in isopret.

I also find the function name ``getLabel`` confusing because it sounds like we are getting the label of the GO term. It would be better to change it to `` T getAnnotatedItem()``

This is what the new interface would look like.
```
public interface TermAnnotation<T> extends Serializable, Comparable<TermAnnotation> {
  TermId getTermId();
  T getAnnotatedItem();
  Optional<String> getEvidenceCode();
  Optional<Float> getFrequency();
}
```

For instance, this could be Arachnodactyly HP:0001166  (The term id) and the item could be [DECIPHER:81](https://hpo.jax.org/app/browse/disease/DECIPHER:81) (in this case, T would be TermId

**Comment by pnrobinson** [comment]
Stale issue, closing.
